src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/UpToDateCheckImplicitConfiguredInput.cs

```diff
@@ -212,8 +212,21 @@ void AddKeys(ImmutableDictionary<string, ImmutableArray<string>> dictionary)
             string? msBuildProjectDirectory = jointRuleUpdate.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.MSBuildProjectDirectoryProperty, MSBuildProjectDirectory);
             string? msBuildProjectOutputPath = jointRuleUpdate.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.OutputPathProperty, OutputRelativeOrFullPath);
             string? outputRelativeOrFullPath = jointRuleUpdate.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.OutDirProperty, msBuildProjectOutputPath);
+            string? projectExtensionsPath = jointRuleUpdate.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.MSBuildProjectExtensionsPathProperty, null);
+            string? nuGetPackageFolders = jointRuleUpdate.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.NuGetPackageFoldersProperty, null);
             string msBuildAllProjects = jointRuleUpdate.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.MSBuildAllProjectsProperty, "");
 
+            // We identify non-modifiable inputs (i.e. anything in Program Files, the VS install dir, or NuGet cache folders)
+            // and exclude them from the set of inputs we scan when an up-to-date query is made.
+            //
+            // For a .NET 5 xUnit project, this cuts the number of file timestamps checked from 187 to 17. Most of those are
+            // reference assemblies for the framework, which clearly aren't expected to change over time.
+            var projectFileClassifier = new ProjectFileClassifier
+            {
+                NuGetPackageFolders = nuGetPackageFolders ?? "",
```

Comment:
> ?? "" [](http://example.com/codeflow?start=57&length=6)

not needed by ProjectFileClassifier

Reply:
The setter throws if it receives null, following a change made after https://github.com/dotnet/project-system/pull/7269#discussion_r640982197.